### DISCUSSION TITLE
Git push Macbook help - Error message password

### DISCUSSION THREAD

**Discussion by lexiroro** [community/community#31627] [New to GitHub] [answered]
On terminal, I enter:

git push -u origin feature/lesson
Username for 'https://github.com': lexiroro
Password for 'https://lexiroro@github.com': 
remote: Support for password authentication was removed on August 13, 2021.
remote: Please see https://docs.github.com/en/get-started/getting-started-with-git/about-remote-repositories#cloning-with-https-urls for information on currently recommended modes of authentication.
fatal: Authentication failed for 'https://github.com/lexiroro/my-first-repo.git/'

I have seen different discussion that I would need to use the Password Authentication Token as my password and that has not worked. Including changing the password in the Keychain Access app with the password authentication token.

**Comment by airtower-luna** [comment, marked as answer]
The PAT approach is correct, you need to input the PAT when Git asks for your password. If it doesn't work there are basically two possibilities:

* The token doesn't have the right scopes, you need **repo** and possibly **workflow** (if you use Actions).
* There's an error when entering the token (e.g. an extra space if you do it by copy & paste).

**Reply by lexiroro** [reply]
Thank you so much. Yes you were right the copy and pasting did not work and just had to manually type it out
